youtube_dl/extractor/redtube.py

```diff
@@ -84,23 +84,33 @@ def _real_extract(self, url):
                 r'mediaDefinition["\']?\s*:\s*(\[.+?}\s*\])', webpage,
                 'media definitions', default='{}'),
             video_id, fatal=False)
-        if medias and isinstance(medias, list):
-            for media in medias:
+        for media in medias if isinstance(medias, list) else []:
+            format_url = url_or_none(media.get('videoUrl'))
+            if not format_url:
+                continue
+            format_id = media.get('format')
+            quality = media.get('quality')
+            if format_id == 'hls' or (format_id == 'mp4' and not quality):
+                more_media = self._download_json(format_url, video_id, fatal=False)
+            else:
+                more_media = [media]
+            for media in more_media if isinstance(more_media, list) else []:
                 format_url = url_or_none(media.get('videoUrl'))
                 if not format_url:
                     continue
-                if media.get('format') == 'hls' or determine_ext(format_url) == 'm3u8':
+                format_id = media.get('format')
+                if format_id == 'hls' or determine_ext(format_url) == 'm3u8':
                     formats.extend(self._extract_m3u8_formats(
                         format_url, video_id, 'mp4',
-                        entry_protocol='m3u8_native', m3u8_id='hls',
+                        entry_protocol='m3u8_native', m3u8_id=format_id if format_id else 'hls',
                         fatal=False))
-                    continue
-                format_id = media.get('quality')
-                formats.append({
-                    'url': format_url,
-                    'format_id': format_id,
-                    'height': int_or_none(format_id),
-                })
+                else:
+                    format_id = media.get('quality')
+                    formats.append({
+                        'url': format_url,
+                        'format_id': format_id,
+                        'height': int_or_none(format_id)
+                    })
```

Comment:
Avoid unrelated changed.

Reply:
It's not really unrelated. The enclosing loop was rewritten and these lines happened to be unchanged apart from indentation. `if...: ...; else: ...` is a better representation of the logic than `if: ...; continue; ...`.

The trailing comma should go back, though.

Reply:
>`if...: ...; else: ...` is a better representation of the logic than `if: ...; continue; ...`. An `elif ...:` might need to be added in future.

That's questionable and the matter of preference. Anyway one should avoid mixing fixes/new features and refactoring in the same commit.

Reply:
You omitted to quote the important point: 

> The enclosing loop was rewritten and these lines happened to be unchanged apart from indentation.

Coding style shouldn't deny "Black mistress" and "cuckold wife" fans their downloads. By all means revert ll. 107 -113 if you wish, but let's get this fixed!

Reply:
To help, I split it into two commits.